packages/block-editor/src/components/block-styles/style.scss

```diff
@@ -43,6 +43,16 @@
 		display: inline-block;
 		width: calc(50% - #{$grid-unit-05});
 
+		&-text {
+			word-break: break-all;
```

Comment:
Great catch, I looked into this a little further and it's pretty interesting. This might be something we need to bake into Truncate. Maybe `word-break: break-all` when `numberOfLines` is 1, and `overflow-wrap: break-word` when it's multiple lines. I'll keep track of this as a separate issue.